ReverseTicTacToe/UI/ConsoleUI.cs

```diff
@@ -33,32 +31,216 @@ public  void Init()
+        public string GetTurnMessage()
+        {
+            string TurnMsg;
+            if((Game.eTurns)m_Game.CurrentTurn == Game.eTurns.PLAYER1)
+            {
+                TurnMsg = "Player 1 your turn, please enter where to place your symbol (row, col): "
+                          + Environment.NewLine;
+            }
+            else
+            {
+                if(m_Game.GameType == Game.eGameTypes.PVP)
+                {
+                    TurnMsg = "Player 2 your turn, please enter where to place your symbol (row, col): "
+                              + Environment.NewLine;
+                }
+                else
+                {
+                    TurnMsg = "The computer is now calculating what's the best next move is:"
+                              + Environment.NewLine;
+                }
+            }
+            return TurnMsg;
+        }
+
+        private void printGameBoard()
+        {
+            StringBuilder boardOutPut = new StringBuilder();
+            int boardLength = m_Game.BoardSize;
+            for (int i = 0; i <= boardLength; i++)
+            {
+                for(int j = 0; j <= boardLength; j++)
+                {
+                    if(i == 0 && j == 0)
+                    {
+                        boardOutPut.Append("  ");
+                    }
+                    else if(i == 0)
+                    {
+                        boardOutPut.Append(j);
+                        boardOutPut.Append("   ");
+                    }
+                    else if(j == 0)
+                    {
+                        boardOutPut.Append(i);
+                    }
+                    else
+                    {
+                        boardOutPut.AppendFormat("| {0} ", m_Game.GameBoard[i - 1, j - 1]);
+                    }
+                }
+
+                if(i > 0)
+                {
+                    boardOutPut.Append("|");
+                }
+
+                boardOutPut.Append(Environment.NewLine);
+                boardOutPut.Append(' ');
+                boardOutPut.Append('=', (boardLength * 4) + 1);
+                boardOutPut.Append(Environment.NewLine);
+            }
+            WriteLine(boardOutPut);
+        }
+
+        private int getBoardSizeFromUser()
+        {
+            string userInputStr = "";
             StringBuilder msg = new StringBuilder();
-            msg.Append("Please enter the board size (choose between: 3/4/5/6/7/8/9)");
+            msg.AppendFormat("Please enter the board size (choose between: {0} - {1})",Game.MINBOARDSIZE,Game.MAXBOARDSIZE);
             msg.Append(Environment.NewLine);
             do
             {
                 WriteLine(msg);
                 userInputStr = ReadLine();
             }
-            while(!isValidInput(userInputStr,1,9));
-            m_BoardSize = int.Parse(userInputStr);
+            while(!isValidInput(userInputStr, Game.MINBOARDSIZE, Game.MAXBOARDSIZE));
+            return int.Parse(userInputStr);
```

Comment:
userInputStr is in format "x,y" right? so isn't using Parse method problematic? because the string has to numbers, and a comma in it

Reply:
userInputStr is not 'x,y' in this case, but a number for the board size.
I will rename the variable to make it more readable and also changed the 'isValidInput' name